snap/snapcraft.yaml

```diff
@@ -32,13 +32,16 @@ parts:
     plugin: go
     source: https://github.com/hashicorp/terraform
     source-type: git
-    source-tag: "v$SNAPCRAFT_PROJECT_VERSION"
+    override-pull: |
+      VERSION="$(craftctl get version)"
+      git clone -b "v$VERSION" https://github.com/hashicorp/terraform $CRAFT_PART_SRC
```

Comment:
@sergiusens this seems to be the right way to solve this in snapcraft 7.0 but feels clunkier - is this correct?